### PR TITLE
improve warning message when values schema is empty

### LINKED ISSUE #703: Tanzu CLI throws an error while getting values schema for the package which does not have any user configurable values.
**Bug description**
Tanzu CLI throws an error while getting values schema for the package which does not have any user configurable values required.
**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [X ] CLI
- [X] Docs
- [X] Installation
- [X] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Expected behavior**
Tanzu CLI should not require data values schema to be non-empty.
**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)
0.4.0
**Environment where the bug was observed (cloud, OS, etc)**
MAC OS
**Relevant Debug Output (Logs, manifests, etc)**
developer-conventions.tanzu.vmware.com package does not have valuesSchema field in package CR.
tanzu package available get developer-conventions.tanzu.vmware.com/0.2.0 --values-schema -n tap-install
Error: data value schema is empty
| Retrieving package details for developer-conventions.tanzu.vmware.com/0.2.0... Usage:
 tanzu package available get PACKAGE_NAME or PACKAGE_NAME/VERSION [flags]

Examples:
  
  # Get package details for a package without specifying the version
  tanzu package available get contour.tanzu.vmware.com --namespace test-ns

  # Get package details for a package with specified version 	
  tanzu package available get contour.tanzu.vmware.com/1.15.1-tkg.1-vmware1 --namespace test-ns

  # Get openAPI schema of a package with specified version
  tanzu package available get contour.tanzu.vmware.com/1.15.1-tkg.1-vmware1 --namespace test-ns --values-schema

Flags:
 -h, --help      help for get
   --values-schema  Values schema of the package, optional

Global Flags:
   --kubeconfig string  The path to the kubeconfig file, optional
   --log-file string   Log file path
 -n, --namespace string  Namespace of packages, optional (default "default")
 -o, --output string    Output format (yaml|json|table), optional
   --verbose int32    Number for the log level verbosity(0-9)

Error: exit status 1

### PR DESCRIPTION
**Pull request by ggpaue**
### What this PR does / why we need it
improve error message when values schema is empty
### Which issue(s) this PR fixes

Fixes #703 

### Describe testing done for PR
manually test on cluster

now if user want to get values schema of a package which which does not have any user configurable values:

tanzu package available get pkg.test.carvel.dev/3.0.0-rc.1 --values-schema -n test-ns --kubeconfig mgmt.config

package 'pkg.test.carvel.dev/3.0.0-rc.1' does not have any user configurable values in the 'test-ns' namespace

### Release note

```release-note

```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: cmd/cli/plugin/package/package_available_get.go [hunk 1]
```diff
@@ -132,6 +132,9 @@ func getValuesSchema(cmd *cobra.Command, args []string, kc kappclient.Client) er
 	}
 
 	var parseErr error
+	if len(pkg.Spec.ValuesSchema.OpenAPIv3.Raw) == 0 {
+		return errors.Errorf("package '%s/%s' does not have any user configurable values in the '%s' namespace", pkgName, pkgVersion, packageAvailableOp.Namespace)
```
Comment (shivaani0505) [inline cmd/cli/plugin/package/package_available_get.go:136]: This should not be an error. Please change it to a warning.

File: cmd/cli/plugin/package/package_available_get.go [hunk 2]
```diff
@@ -132,6 +132,9 @@ func getValuesSchema(cmd *cobra.Command, args []string, kc kappclient.Client) er
 	}
 
 	var parseErr error
+	if len(pkg.Spec.ValuesSchema.OpenAPIv3.Raw) == 0 {
```
Comment (shyaamsn) [inline cmd/cli/plugin/package/package_available_get.go:135]: Could you please add a unit-test to cover this case.

File: cmd/cli/plugin/package/package_available_get.go (hunk 1 above)
Comment (ggpaue) [reply cmd/cli/plugin/package/package_available_get.go]: changed to warning

File: cmd/cli/plugin/package/package_available_get.go (hunk 2 above)
Comment (danniel1205) [reply cmd/cli/plugin/package/package_available_get.go:135]: unit tests will be added by https://github.com/vmware-tanzu/tanzu-framework/pull/1056

Comment (danniel1205) [reply cmd/cli/plugin/package/package_available_get.go:135]: suggestion: `ValuesSchema` and `OpenAPIv3` are optional fields and they are both nullable, that means `pkg.Spec.ValuesSchema` or `pkg.Spec.ValuesSchema.OpenAPIv3` could be `nil`. We might want to add more checks from line#135

**Review by danniel1205 — APPROVED** [review]
LGTM

**Pull request merged**